src/main/scala/chisel3/util/circt/IsX.scala

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.{annotate, ChiselAnnotation, ExtModule}
+
+import circt.Intrinsic
+
+// We have to unique designedName per type, be we can't due to name conflicts
+// on bundles.  Thus we use a globally unique id.
+private object IsXGlobalIDGen {
```

Comment:
@jackkoenig What happened to putting this cache into the builder?

Reply:
@darthscsi I agree with Kevin here, please do what I suggested here: https://github.com/chipsalliance/chisel3/pull/2900#discussion_r1085901366

All of these ids could live in a single mutable HashMap that keys by their "id namespace"

Reply:
I've reused an existing mechanism now.  It would be nice if one could use instantiate directly, but that is limited to RawModule derived types.

Reply:
@jackkoenig is talking about something different. This is only about how unique IDs are generated in the case where multiple Chisel circuits are elaborated in parallel in the same process. It would probably best if the two of you had a short chat to sort this out. As @mwachs5 mentioned, the current approach could lead to non-stable Chirrtl output in the case of parallel elaborations.